spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe "customers Show Page" do
+ before(:each) do
+  @customer_1 = Customer.create!(name: "Billy")
+    @supermarket_1 = Supermarket.create!(name: "Sals Super Market", location: "321 Grocery Grove")
+    @item_1 = @supermarket_1.items.create!(name: "Beef", price: 5)
+    @item_2 = @supermarket_1.items.create!(name: "Cheese", price: 3)
+    @item_3 = @supermarket_1.items.create!(name: "Buns", price: 1)
+    @item_4 = @supermarket_1.items.create!(name: "Cherries", price: 2)
+    
+    CustomerItem.create!(item_id: @item_1.id, customer_id: @customer_1.id)
+    CustomerItem.create!(item_id: @item_3.id, customer_id: @customer_1.id)
+ end
+ 
+ 
+  #User Story 1
+  it "shows the customer's name and a lists of items" do
+    
+    visit "/customers/#{@customer_1.id}"
+    
+    expect(page).to have_content(@customer_1.name)
+    expect(page).to have_content(@supermarket_1.name)
+    expect(page).to have_content(@item_1.name)
+    expect(page).to have_content(@item_1.price)
+    expect(page).to have_content(@item_3.name)
+    expect(page).to have_content(@item_3.price)
+    expect(page).to_not have_content(@item_2.name)
+    expect(page).to_not have_content(@item_2.price)
```

Comment:
Good job testing all of the info we wanted to see on this page. I especially like that you included a to_not have_content test to ensure this wasn't just showing ALL the items. I do agree with your reflection that adding within blocks is going to help with targeted testing. Right now you're testing all this content is on the page, but the item_1 price might be mixed up in a section with the item_3 stuff, we don't know! So within blocks will help being more explicit and specific.